addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -885,6 +886,84 @@ def test_load_pos_data_should_not_fail(self):
+        +---------------------+---------+
+        | Total balance       |     0.0 |
+        +---------------------+---------+
+        """
+        def _before_closing_cb():
+            # return order
+            order_to_return = self.pos_session.order_ids.filtered(lambda order: '12345-123-1234' in order.pos_reference)
+            order_to_return.refund()
+            refund_order = self.pos_session.order_ids.filtered(lambda order: order.state == 'draft')
+
+            # check if amount to pay
+            self.assertAlmostEqual(refund_order.amount_total - refund_order.amount_paid, -30)
+
+            # pay the refund
+            context_make_payment = {"active_ids": [refund_order.id], "active_id": refund_order.id}
+            make_payment = self.env['pos.make.payment'].with_context(context_make_payment).create({
+                'payment_method_id': self.cash_pm1.id,
+                'amount': -30,
+            })
+            make_payment.check()
+
+        self._run_test({
+            'payment_methods': self.cash_pm1 | self.bank_pm1,
+            'orders': [
+                {'pos_order_lines_ui_args': [(self.product3, 1)], 'payments': [(self.cash_pm1, 30)], 'uid': '12345-123-1234'},
+            ],
+            'before_closing_cb': _before_closing_cb,
+            'journal_entries_before_closing': {},
+            'journal_entries_after_closing': {
+                'session_journal_entry': {
+                    'line_ids': [
+                        {'account_id': self.sales_account.id, 'partner_id': False, 'debit': 0, 'credit': 30, 'reconciled': False},
+                        {'account_id': self.sales_account.id, 'partner_id': False, 'debit': 30, 'credit': 0, 'reconciled': False},
+                    ],
+                },
+                'cash_statement': [],
+                'bank_payments': [],
+            },
+        })
+
+        # keep reference of the closed session
+        closed_session = self.pos_session
+        self.assertTrue(closed_session.state == 'closed', 'Session should be closed.')
+
+        return_to_invoice = closed_session.order_ids[1]
+        test_customer = self.env['res.partner'].create({'name': 'Test Customer'})
+
+        with freeze_time(fields.Datetime.now() + relativedelta(days=2)):
+            # create new session after 2 days
+            self.open_new_session(0)
+            # invoice the uninvoiced order
+            return_to_invoice.write({'partner_id': test_customer.id})
+            return_to_invoice.action_pos_order_invoice()
+            # check invoice
+            self.assertTrue(return_to_invoice.account_move, 'Invoice should be created.')
+            self.assertEqual(return_to_invoice.account_move.move_type, 'out_refund', 'Invoice should be a credit note.')
+            self.assertTrue(return_to_invoice.account_move.invoice_date != return_to_invoice.date_order.date(), 'Invoice date should not be the same as order date since the session was closed.')
```

Comment:
- use `assertRecordValues`
- don't make test depending of today. Instead, put an arbitrary date.
- Instead of comparing the date is different, compare with the expected exact value.
- assert the invoice lines and at least, the total amount_untaxes/tax/total

Reply:
Okay
- Checking the records
- Using `order_date` + 2 as a variable now and comparing that value
- Checking the totals of the invoice as well